susemanager-utils/susemanager-sls/salt/distupgrade/init.sls

```diff
@@ -13,5 +13,16 @@ spmigration:
         - file: mgrchannels*
 {% endif %}
 
+{% if not salt['pillar.get']('susemanager:distupgrade:dryrun') %}
+{% if pillar.get('missing_successors', [])%}
+mgr_release_pkg_removed:
+  pkg.removed:
+    -   pkgs:
+{%- for missing_successor in pillar.get('missing_successors', [])%}
+        - {{missing_successor}}-release
+{%- endfor %}
+{% endif %}
+{% endif %}
+
```

Comment:
Maybe it would be better to executed this "before" the distupgrade.

Maybe some of these release packages would prevent a successful upgrade when the release packages without successor are still available.

Reply:
I wanted to let zypper do the cleanup what it can and then the remaining cleanup will be done by us just to be sure.

Like in the case of LTSS, zypper display pretty nice log info about removing LTSS product. If we will remove the products by us 1st, user wouldn't be seeing such information in upgrade logs or in dry run. Anyways, this is the usability part that I had in mind.
I can change it if you think we better remove such products 1st before running the SP migration. I will need to do some testing though.

Reply:
ok